Dockerfile

```diff
@@ -1,18 +1,35 @@
-FROM quay.io/app-sre/fluentd-upstream:v1.16-1
+FROM registry.redhat.io/openshift-logging/fluentd-rhel9@sha256:a31038fa2b953660e09b7db8411e808075c9037b679ba56a4a6245f56853db56
 
 USER root
 
-RUN apk update && apk upgrade busybox busybox-binsh libcrypto3 libssl3 \
-    ncurses-libs ncurses-terminfo-base ssl_client \
-    && apk add --no-cache --update --virtual .build-deps build-base ruby-dev \
-    && echo 'gem: --no-document' >> /etc/gemrc \
-    && gem install fluent-plugin-s3 \
-    && gem install fluent-plugin-slack \
-    && gem install fluent-plugin-cloudwatch-logs \
-    && gem install fluent-plugin-teams \
-    && gem install fluent-plugin-rewrite-tag-filter \
-    && gem install nokogiri \
-    && apk del .build-deps \
-    && rm -rf /tmp/* /var/tmp/* /usr/lib/ruby/gems/*/cache/*.gem
+RUN microdnf install -y \
+    gcc \
+    make \
+    ncurses-base \
+    ncurses-libs \
+    redhat-rpm-config \
+    ruby-devel \
+    openssl \
+    openssl-libs \
+ && microdnf clean all
 
+# skip installing documentation
+RUN echo 'gem: --no-document' >> /etc/gemrc
+
+RUN curl -L https://busybox.net/downloads/binaries/1.36.0-defconfig-multiarch/busybox-x86_64 \
+    -o /usr/local/bin/busybox && \
+    chmod +x /usr/local/bin/busybox && \
+    ln -s /usr/local/bin/busybox /usr/local/bin/sh
+
+RUN gem install \
+    fluent-plugin-s3 \
+    fluent-plugin-slack \
+    fluent-plugin-cloudwatch-logs \
+    fluent-plugin-teams \
+    fluent-plugin-rewrite-tag-filter \
+    nokogiri && \
+    rm -rf /usr/share/gems/cache/*.gem /tmp/* /var/tmp/*
```

Comment:
versions not pinned, we can put all of them plus fluentd in a `Gemfile`, example https://docs.fluentd.org/deployment/plugin-management#gemfile-option, note we don't need to use fluentd to install gems at runtime, we can do it at build time.

Reply:
@hemslo I think the Gemfile idea is fine, but do we actually want to pin the versions? 

I think we avoid this for most of our simpler Dockerfiles to make it easier to automate regular builds with up-to-date dependencies, and then deploy with those new images to remediate CVEs.

Reply:
Updated this MR with both a Gemfile and the current versions of the plugins that we have in our current Dockerfile